src/main/java/no/sikt/sws/IndexHandler.java

```diff
@@ -40,24 +42,20 @@ protected ProxyResponse<String> processProxyInput(
             Context context
     ) throws ApiGatewayException {
 
-        var query = EMPTY_STRING;
-        try {
-            var uri = request.getRequestUri();
-            if (!uri.getQuery().isBlank()) {
-                query = "?" + uri.getQuery();
-            }
-        } catch (Exception ex) {
-            logger.info(ex.getMessage());
-        }
         var resourceIdentifier =  request.getPathParameter(RESOURCE_IDENTIFIER);
         var httpMethod = RequestUtil.getRequestHttpMethod(request);
         var workspace = RequestUtil.getWorkspace(request);
         var openSearchCommand = OpenSearchCommand.fromString(resourceIdentifier);
+        var query = request.getQueryParameters().entrySet().stream()
+            .map(entry ->
+                entry.getKey() + "=" + entry.getValue()).collect(Collectors.joining("&"));
 
         try {
             validateResourceIdentifier(resourceIdentifier);
 
-            var url = Prefixer.url(workspace, resourceIdentifier) + query;
+            var url = Prefixer.url(
+                workspace,
+                resourceIdentifier + (query.isEmpty() ? EMPTY_STRING : "?" + query));
```

Comment:
I noticed that the query is now called inside Prefixer.url() but before it was outside. Is this intented, and needed? or an accident?

Also 
`(query.isEmpty() ? EMPTY_STRING : "?" + query `
should be stored as a variable